pkgs/applications/networking/wgnord/default.nix

```
@@ -0,0 +1,67 @@
+{ bash
+, coreutils
+, curl
+, fetchFromGitHub
+, gnugrep
+, gnused
+, iproute2
+, jq
+, lib
+, makeWrapper
+, resholve
+, wireguard-tools
+}:
+
+resholve.mkDerivation rec {
+  pname = "wgnord";
+  version = "0.1.10";
+
+  src = fetchFromGitHub {
+    owner = "phirecc";
+    repo = pname;
+    rev = version;
+    hash = "sha256-T7dAEgi4tGvrzBABGLzKHhpCx0bxSCtTVI5iJJqJGlE=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postPatch = ''
+    substituteInPlace wgnord \
+      --replace '$conf_dir/countries.txt' "$out/share/countries.txt" \
+      --replace '$conf_dir/countries_iso31662.txt' "$out/share/countries_iso31662.txt"
+  '';
+
+  dontBuild = true;
+
+  installPhase = ''
+    install -Dm755 wgnord -t $out/bin/
+    install -Dm 644 countries.txt -t $out/share/
+    install -Dm 644 countries_iso31662.txt -t $out/share/
+  '';
+
+  solutions.default = {
+    scripts = [ "bin/wgnord" ];
+    interpreter = "${bash}/bin/bash";
+    inputs = [
+      coreutils
+      curl
+      gnugrep
+      gnused
+      iproute2
+      jq
+      wireguard-tools
+    ];
+    execer = [
```

```suggestion
    fix.aliases = true; # curl command in an alias
    execer = [
```

This one was also in my diff. Aliases are a bit unfortunate. When they're in a user's actual shell profile, resolving them generally isn't right. In a script, they often should be. 

I guess it might be safer, if a little pickier, if we flip it around: always check the aliases for commands, and require consent for *either* outcome if commands were found in them.

---

I tend to treat all applications as black boxes and my goal with this whole process, especially the move to resholve, is to be explicit about dependencies; thus if you want a shell alias to propagate into wgnord, I would expect a user would use a derivation to do it: (or an officially exposed flag or env var by the upstream tool)
```nix
pkgs.wgnord.override { curl = curlWithGnuTls; }
```

---

Not sure if we're on same page. In this case I mean the ones defined in wgnord:

```bash
alias query="curl -s -H 'User-Agent: NordApp Linux 3.12.0 Linux 5.4.0-58-generic'"
```

from https://github.com/phirecc/wgnord/blob/79c4754f9564e6e8e6a3723d2d37d08b67abd2ea/wgnord#L2

So the resholve directive is just saying, "yes, parse aliases to resolve into them".

---

yep, I had my wires crossed; and yeah, i like the idea of parsing aliases by default, but happy to do that explicitly